irods/test/data_obj_test.py

```diff
@@ -1959,6 +1960,41 @@ def test_settings_load_and_save_471(self):
                     config.load(**load_logging_options)
                     self.assertTrue(config.data_objects.auto_close, RANDOM_VALUE - i - 1)
 
+    def test_append_mode_will_append_to_data_object__issue_495(self):
+        append_string = b'to_be_written'.lower()
+        reverse_bytes = lambda s: ''.join(reversed(s)) if six.PY2 else bytes(reversed(s))
+        session, data = (self.sess, self.sess.data_objects)
+        testfile = '{}/issue_495'.format(helpers.home_collection(session))
+        # Make sure data object doesn't exist.
+        self.assertFalse(data.exists(testfile))
+        try:
+            # Append to a previously nonexistent data object.
+            with data.open(testfile,'ab') as f:
+                f.write(append_string.upper())
+            # Test that the data object, once opened, has the right offset. Then perform a second append.
+            with data.open(testfile,'a+b') as f:
+                self.assertTrue(f.tell(), len(append_string))
+                f.write(append_string)
+            # Verify original content written at offset 0.
+            with data.open(testfile,'rb') as f:
+                self.assertEqual(f.read(), append_string.upper())
```

Comment:
why does this pass?   shouldn't it read() two-lengths of string, one upper, and one not?  but only matching against a single-length-upper?

Reply:
You were indeed correct.  And... I had 'except' instead of finally, *and* 'b' is not allowed in a mode string in PRC, which was causing the first open to quit the test section entirely!

Reply:
Test works now, corrected as agreed upon.